lua/wire/stools/expression2.lua

```diff
@@ -222,6 +220,7 @@ if SERVER then
 
 	function WireLib.Expression2Upload( ply, target, filepath )
 		if not IsValid( target ) then error( "Invalid entity specified" ) end
+		target.AwaitingUpload = true
```

Comment:
TOOL:Upload looks like it easily is spamable so AwaitingUpload will be easily set to true. Should 'return' if target.Uploading is true.